charts/newrelic-infrastructure-beta/templates/_helpers.tpl

```diff
@@ -0,0 +1,190 @@
+*/}}
+{{- define "newrelic.licenseCustomSecretName" -}}
+{{- if .Values.customSecretName -}}
+  {{- .Values.customSecretName -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.customSecretName -}}
+    {{- .Values.global.customSecretName -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name of the secret holding the License Key
+*/}}
+{{- define "newrelic.licenseSecretName" -}}
+{{ include "newrelic.licenseCustomSecretName" . | default (printf "%s-license" (include "newrelic.fullname" . )) }}
+{{- end -}}
+
+{{/*
+Return the name key for the License Key inside the secret
+*/}}
+{{- define "newrelic.licenseCustomSecretKey" -}}
+{{- if .Values.customSecretLicenseKey -}}
+  {{- .Values.customSecretLicenseKey -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.customSecretLicenseKey }}
+    {{- .Values.global.customSecretLicenseKey -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name key for the License Key inside the secret
+*/}}
+{{- define "newrelic.licenseSecretKey" -}}
+{{ include "newrelic.licenseCustomSecretKey" . | default "licenseKey" }}
+{{- end -}}
+
+{{/*
+Returns nrStaging
+*/}}
+{{- define "newrelic.nrStaging" -}}
+{{- if .Values.nrStaging -}}
+  {{- .Values.nrStaging -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.nrStaging -}}
+    {{- .Values.global.nrStaging -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns fargate
+*/}}
+{{- define "newrelic.fargate" -}}
+{{- if .Values.fargate -}}
+  {{- .Values.fargate -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.fargate -}}
+    {{- .Values.global.fargate -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the list of namespaces where secrets need to be accessed by the controlPlane integration to do mTLS Auth
+*/}}
+{{- define "newrelic.roleBindingNamespaces" -}}
+{{ $namespaceList := list }}
+{{- range $components := .Values.controlPlane.config }}
+    {{- range $autodiscover := $components.autodiscover }}
+        {{- range $endpoint := $autodiscover.endpoints }}
+            {{- if $endpoint.auth }}
+            {{- if $endpoint.auth.mtls }}
+            {{- if $endpoint.auth.mtls.secretNamespace }}
+            {{- $namespaceList = append $namespaceList $endpoint.auth.mtls.secretNamespace -}}
+            {{- end }}
+            {{- end }}
+            {{- end }}
+        {{- end }}
+    {{- end }}
+{{- end }}
+roleBindingNamespaces: {{- uniq $namespaceList | toYaml | nindent 0 }}
+{{- end -}}
+
+{{/*
+Returns Custom Attributes even if formatted as a json string
+*/}}
+{{- define "newrelic.customAttributesWithoutClusterName" -}}
+{{- if kindOf .Values.customAttributes | eq "string" -}}
+{{  .Values.customAttributes }}
+{{- else -}}
+{{ .Values.customAttributes | toJson }}
+{{- end -}}
+{{- end -}}
+
+{{- define "newrelic.customAttributes" -}}
+{{- merge (include "newrelic.customAttributesWithoutClusterName" . | fromJson) (dict "clusterName" "$(CLUSTER_NAME)") | toJson | quote }}
```

Comment:
With `$(CLUSTER_NAME)` Kubernetes is using other environment variable from the pod itself. There is no pod with `CLUSTER_NAME` variable so it is going to be empty. Use the exist of `{{ include "newrelic.cluster" . }}`